### PR TITLE
Local llm mutli-gpu support

### PR DESCRIPTION
**Pull request by rounak610**
### Description

Added multi-gpu support in local llms feature
Command to build without local llms: docker compose -f docker-compose.yaml up --build  
Command to build with local llms: docker compose -f docker-compose-gpu.yml up --build 

### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have read the contributing guide and my code conforms to the guidelines.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: config_template.yaml [hunk 1]
```diff
@@ -122,3 +122,4 @@ ENGINE_ID: "stable-diffusion-xl-beta-v2-2-2"
 ## To use Qdrant for vector store
 #QDRANT_HOST_NAME: YOUR_QDRANT_HOST_NAME
 #QDRANT_PORT: YOUR_QDRANT_PORT
+GPU_LAYERS: GPU LAYERS THAT YOU WANT TO OFFLOAD TO THE GPU WHILE USING LOCAL LLMS
```
Comment (Fluder-Paradyne) [inline config_template.yaml:125]: comment this out

File: docker-compose.yaml [hunk 2]
```diff
@@ -3,18 +3,23 @@ services:
   backend:
     volumes:
       - "./:/app"
-    build: .
+    build:
```
Comment (Fluder-Paradyne) [inline docker-compose.yaml:6]: why this change ?

Comment (Fluder-Paradyne) [reply docker-compose.yaml:6]: this change doesnt change any behavior

File: superagi/helper/llm_loader.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def model(self):
         if self._model is None:
             try:
                 self._model = Llama(
-                    model_path="/app/local_model_path", n_ctx=self.context_length)
+                    model_path="/app/local_model_path", n_ctx=self.context_length, n_gpu_layers=get_config('GPU_LAYERS'))
```
Comment (Fluder-Paradyne) [inline superagi/helper/llm_loader.py:25]: give default

File: Dockerfile [hunk 4]
```diff
@@ -13,6 +13,7 @@ ENV PATH="/opt/venv/bin:$PATH"
 COPY requirements.txt .
 RUN pip install --upgrade pip && \
     pip install --no-cache-dir -r requirements.txt
+RUN python3 -m pip install llama-cpp-python==0.2.7 --force-reinstall --upgrade --no-cache-dir
```
Comment (Fluder-Paradyne) [inline Dockerfile:16]: test

**Review by Fluder-Paradyne — APPROVED** [review]

**Review by Lexi4202132 — COMMENTED** [review]
Thanks
